driverapi/driverapi.go

```diff
@@ -106,3 +104,107 @@ type DriverCallback interface {
 type Capability struct {
 	DataScope datastore.DataScope
 }
+
+// IPData represents the per-network ip related
+// operational information libnetwork will send
+// to the network driver during CreateNetwork()
+type IPData struct {
+	AddressSpace string
+	Pool         *net.IPNet
+	Gateway      *net.IPNet
+	Addresses    map[string]*net.IPNet
+}
+
+// MarshalJSON encodes IPData into json message
+func (i *IPData) MarshalJSON() ([]byte, error) {
```

Comment:
Is it okay to move these Marshalling routines to a different file and keep the driverapi.go for API definition ?